### PR TITLE
cr/sunrise

### PR DESCRIPTION
**Pull request by Carlos-oficial**

### CODE REVIEW

File: content.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  "Experimentação em Engenharia de Software": "EES",
+  "Gestão e Segurança de Redes": "GSR",
+  "Redes Definidas por Software": "RDS",
+  "Sistemas Distribuídos em Grande Escala": "SDGE",
+  "Verificação Formal": "VF",
+  "Visão por Computador e Processamento de Imagem": "VCPI",
+  "Administração de Bases de Dados": "ABD",
+  "Aprendizagem Profunda": "AProf",
+  "Bases de Dados NoSQL":
+    "short_n  NoSQL",
```
Comment (ruilopesm) [inline content.js]: Why is this line like so?

Comment (Carlos-oficial) [reply content.js]: no hablo inglés

File: manifest.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "manifest_version": 2,
+  "name": "Swapium",
+  "version": "1.0",
+
+  "description": "Interface entre o Swap e Calendarium",
```
Comment (diogogmatos) [inline manifest.json]:
```suggestion
  "description": "🔄 Connecting SWAP to Calendarium ",
```

File: manifest.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "manifest_version": 2,
+  "name": "Swapium",
```
Comment (diogogmatos) [inline manifest.json]:
```suggestion
  "name": "SWAPIUM",
```

File: content.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+button.onclick = () => {
+  groupedByCourse = parsePage();
+  shiftString = ""
+  Object.keys(groupedByCourse).map(name => {
+    const shortName = getClosestShortName(name);
+    shiftString += (`${shortName}=${groupedByCourse[name]}&`);
+  });
+  shiftString = shiftString.slice(0,-1)
+  navigator.clipboard.writeText(shiftString);
+  alert('Copied the text: ' + shiftString);
```
Comment (diogogmatos) [inline content.js]:
```suggestion
  if (shiftString !== '' && window.confirm('The share code was generated and copied successfully. Click the 'Share' icon in Calendarium and paste the code to proceed.'))
  {
    window.open('https://calendario.cesium.pt', '_blank');
  }
  else
  {
    alert('Something went wrong. Please make sure you are logged in.');
  }
```

This allows us to show an alert window with 'Ok' and 'Cancel' buttons, when clicking 'Ok' Calendarium opens automatically in a new window.

https://stackoverflow.com/questions/16973240/link-in-alert-boxes-javascript
https://www.geeksforgeeks.org/how-to-open-url-in-new-tab-using-javascript/

**EDIT:** Also, we could alert the user if the string was not generated successfully, like when the user is not logged in and an empty string is generated.

**Review by diogogmatos — APPROVED** [review]

**Pull request merged**